build-scripts/ubuntu-2004/prepare-package.sh

```diff
@@ -33,6 +33,7 @@ if [ "$distro_packages" = "debian-packages" ]; then
   echo -e "\nAdapt the dependencies for the Canonical archive"
   sed -i "s~timeout-decorator~python3-timeout-decorator~" setup.py
   sed -i "s~distro~python3-distro~" setup.py
+  sed -i "s~importlib-metadata~python3-importlib-metadata~" setup.py
```

Comment:
@mgmgwi could you remove this line? 
We don't use `python3-importlib-metadata` from the Canonica archive because `python3-importlib-metadata` is still at version `1.5.0-1` there (https://packages.ubuntu.com/focal/python3-importlib-metadata)

We build the Debian package based on the PyPi packages of `python3-importlib-metadata`, publish it to Hyperledger's Artifactory and pull the artifact from there.

Reply:
Hmm, the problem is, that the .deb package contains a reference to `importlib-metadata` which we have to replace or remove. If not here, where then?

Reply:
The dependencies are taken from `setup.py`. `importlib-metadata==3.10.1` is referenced there already. 
When installing the Debian package of Indy-Node `dpgk` looks for `python3-importlib-metadata==3.10.1` which will be pulled from here https://hyperledger.jfrog.io/ui/native/indy/pool/focal/dev/i/importlib-metadata/

Reply:
:thinking: are you sure? I think it will look for `importlib-metadata=3.10.1` (missing the `python3-` prefix) which still does not exists. I tried manually patching the control file in the .deb package like this:

```
Package: indy-node
Version: 1.13.0~dev24
License: Apache 2.0
Vendor: none
Architecture: amd64
Maintainer: Hyperledger <hyperledger-indy@lists.hyperledger.org>
Installed-Size: 2807
Depends: indy-plenum (= 1.13.0~dev14), importlib-metadata (= 3.10.1), python3-timeout-decorator (>= 0.5.0), python3-distro (>= 1.5.0), at, iptables, libsodium23
Section: default
Priority: extra
Homepage: https://github.com/hyperledger/indy-node
Description: Indy node
```
Please note the `Depends` line. Then I repackaged it, but when I install it with `apt install /tmp/indy-node_1.13.0_dev24-patched2_amd64.deb` it still complains:

```
The following packages have unmet dependencies:
 indy-node : Depends: importlib-metadata (= 3.10.1) but it is not installable
```

Reply:
I'm quite sure but I cannot verify before Tuesday. 
We use `fpm` to build the Debian artifacts with the following flag `--python-package-name-prefix "python3"`
All `install_requires` (https://github.com/hyperledger/indy-node/blob/ubuntu-20.04-upgrade/setup.py#L58) packages will be converted to `python3-package_name`. 
Therefore we publish the third-party dependencies with the `python3` prefix. And in case we use the artifacts from the canonical archive we adjust them to the correct build version. An example for Indy Plenum can be found [here ](https://github.com/hyperledger/indy-plenum/blob/master/build-scripts/ubuntu-1604/prepare-package.sh#L31) for Ubuntu 26.04